src/commons/api/rrdp.rs

```diff
@@ -731,3 +734,79 @@ impl DeltaData {
         writer.done()
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use crate::test::*;
+
+    #[test]
+    fn current_objects_delta() {
```

Comment:
Where does this test verify the correct behaviour for the scenario described in #981? I would expect to see a test here that attempts to publish an object with URI A and hash X and then to publish an object with URI A and hash Y, i.e. verify that a client that _"sends a publish element for an existing uri" (ed: slightly paraphrased from what was said on #981) will not succeed. I assume that the `fileN_content` variables contain hashes rather than file content, and thus that I should be looking for a test that for example publishes `file1_uri` with `file1_content` and then attempts to publish `file1_uri` with `file2_content`, but I only see a single publish followed by an identical publish rather than a same-by-uri-but-not-hash publish, and then after that only updates and withdraws.